src/firecracker/src/main.rs

```diff
@@ -281,6 +293,30 @@ fn print_supported_snapshot_versions() {
     println!("{}\n", snapshot_versions_str);
 }
 
+// Print data format of provided snapshot state file.
+fn print_snapshot_data_format(snapshot_path: &str) {
+    let mut snapshot_reader =
```

Comment:
I do not think we want to panic if the file containing the state is invalid. I suggest to add logic that exits gracefully. You can add an auxiliary function to avoid duplicated code:
```
fn generic_error_exit(msg: &str) -> u8 {
    error!("{}", msg);
    return vmm::FC_EXIT_CODE_GENERIC_ERROR;
}
```
and
```
let mut snapshot_reader = File::open(snapshot_path).unwrap_or_else(|err| {
        process::exit(i32::from(generic_error_exit(&format!(
            "Unable to open snapshot state file: {}",
            err
        ))));
    });
    let data_format_version = Snapshot::get_data_version(&mut snapshot_reader, &VERSION_MAP)
        .unwrap_or_else(|err| {
            process::exit(i32::from(generic_error_exit(&format!(
                "Invalid data format version of snapshot file: {:?}",
                err
            ))));
        });
```